clientv3/concurrency/mutex.go

```diff
@@ -80,14 +81,28 @@ func (m *Mutex) Lock(ctx context.Context) error {
 	}
 	client := m.s.Client()
 	// wait for deletion revisions prior to myKey
+	// TODO: early termination if the session key is deleted before other session keys with smaller revisions.
 	hdr, werr := waitDeletes(ctx, client, m.pfx, m.myRev-1)
 	// release lock key if wait failed
 	if werr != nil {
 		m.Unlock(client.Ctx())
 	} else {
 		m.hdr = hdr
 	}
-	return werr
+
+	// make sure the session is not expired, and the owner key still exists.
+	gresp, werr := client.Get(ctx, m.myKey)
+	if werr != nil {
+		m.Unlock(client.Ctx())
+		return werr
+	}
+
+	m.hdr = gresp.Header
```

Comment:
Swapping the assignment of `hdr` and checking the number of keys would be safer? In other places users won't see initialized `hdr` when locking is failed.